modules/artifact/src/main/java/org/mule/runtime/module/artifact/classloader/MuleDeployableArtifactClassLoader.java

```diff
@@ -36,7 +36,27 @@ public MuleDeployableArtifactClassLoader(String artifactId, ArtifactDescriptor a
                                            ClassLoader parent,
                                            ClassLoaderLookupPolicy lookupPolicy,
                                            List<ArtifactClassLoader> artifactPluginClassLoaders) {
-    super(artifactId, artifactDescriptor, urls, parent, lookupPolicy);
+    this(artifactId, artifactDescriptor, urls, parent, lookupPolicy, artifactPluginClassLoaders, true);
+  }
+
+  /**
+   * Creates a {@link MuleDeployableArtifactClassLoader} with the provided configuration.
+   *
+   * This version allows not registering error hooks, useful for subclasses that need to do it after
```

Comment:
We should not talk about "error hooks" in the API. That is something that has to be kept internal, so we can remove it in the future